### PR TITLE
[Plugins] Bring over Timeline, Clocks/Timers

### PR DESCRIPTION
**Pull request by VWoeltjen**
Brings over plugins from closed-source repository for clocks/timers.

Summary of changes:

* Copy over plugins.
* De-namespace type identifiers, directives, etc.
* Disabled timelines due to missing styles

Plan to file follow up issues for:
- [x] ~~Bringing over timeline styles.~~ #208
- [x] ~~Updating code style to use prototypes for classes in Timeline~~ #210
- [x] ~~Updating code style to use prototypes for classes in Clocks/Timers~~ #209
- [x] ~~Migrating existing databases which used prefixed type keys~~ WTD-1562

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Expect to pass code review? Y

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Build failing? Reassigning to me to figure out why

**Comment by VWoeltjen** [comment]
`moment-duration-format` has a dependency on `moment` which wasn't captured in require configuration - this created some timing/ordering sensitivity in the test suite, since `moment-duration-format` would still work if it happened to be loaded first. Added some require configuration to make this dependency explicit.

Command line build (`mvn clean install`) now succeeding regularly; trying to figure out why `npm test` is not.

**Comment by larkin** [comment]
@VWoeltjen `npm test` doesn't currently read requirejs configuration from bundles, so requirejs configuration that is present in bundles.json must be manually added to test configuration in `test-main.js`.

**Comment by VWoeltjen** [comment]
Thanks @larkin - added the relevant config to `test-main.js` and removed a (now unecessary) inclusion of `**/moment*` from karma's conf. `npm test` completes happily now.

**Comment by akhenry** [comment]
I am seeing some errors affecting compositing in this branch. 
eg.
1. Create a new Layout
2. Using the create button, add a sine plot to the layout.

In the javascript console I am seeing an error from the GenericSearchProvider -

    TypeError: Cannot set property 'de7305b2-e966-4e58-a2bc-c18606d8c230' of undefined
        at http://localhost:8080/platform/search/src/services/GenericSearchProvider.js:125:34

When I reload the page, the sine plot is missing from the layout.

**Comment by akhenry** [comment]
Reassigning for investigation

**Comment by VWoeltjen** [comment]
This looks like a separate bug that was resolved in master by #205; merged in latest from master branch into this branch and no longer see the issue. (Also resolved merge conflicts in test scripts.)

File: platform/features/clock/src/controllers/RefreshingController.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+define(
+    [],
+    function () {
+        "use strict";
+
+        /**
+         * Continually refreshes the represented domain object.
+         *
+         * This is a short-term workaround to assure Timer views stay
```
Comment (akhenry) [inline platform/features/clock/src/controllers/RefreshingController.js]: Is this workaround something that we need to create a separate issue to address?

**Comment by akhenry** [comment]
New source files need license text (eg. ClockController.js, AbstractStartTimerAction.js, StartTimerAction.js, etc.)

Have only done a very perfunctory code review of the 122 source files, assuming more detailed code review was done at the time they were committed to WARP. If not, suggest we divide up the reviewing between us, as well as any remedial work necessary.

**Comment by VWoeltjen** [comment]
Yep, these changes have all undergone the same review process in closed-source, don't think we need to repeat that here.

Thanks for catching the missing license headers, I'll add those

**Comment by VWoeltjen** [comment]
@akhenry  - License headers have been added

**Comment by akhenry** [comment]
Looks good now.

I do note that styles are missing for timelines, but there is an issue for this (#208).

# Reviewer Checklist

* Changes appear to address issue? __Y__
* Appropriate unit tests included? __Y *__
* Code style and in-line documentation are appropriate? __Y *__
* Commit messages meet standards? __Y__

__*__ Code was reviewed on commit to WARP.

**Pull request merged**